src/bitcoin-cli.cpp

```diff
@@ -259,6 +259,14 @@ static void http_error_cb(enum evhttp_request_error err, void *ctx)
     reply->error = err;
 }
 
+static int8_t NetworkStringToId(const std::string& str)
```

Comment:
from the usages it seems we can modernize this in a few more ways:
* `UNKNOWN_NETWORK` seems to serve as a value-missing proxy - we could use an optional instead
* `const std::string&` could be modernized to `static std::optional<int8_t> NetworkStringToId(std::string_view network)`


This would allow us to change the usages to something like:
```C++
if (auto network_id{NetworkStringToId(network_name)}) {
    ++counts.at(*network_id);
}
```

Reply:
Yes, IIRC `std::optional` wasn't available when this was added, as the codebase was using C++11 at that time. I think this would be more of a modernization rewrite than a cleanup, but happy to review if you want to propose it.